client/components/course/Sidebar/Relationship.vue

```diff
@@ -55,14 +59,29 @@ export default {
         const lineage = this.getLineage(this.activity);
         conds.push(it => !includes(lineage, it));
       }
+      if (this.allowedTypes.length) {
+        conds.push(({ type }) => {
+          const { typeName } = parseType(type);
+          return includes(this.allowedTypes, typeName);
+        });
+      }
       return filter(activities, it => every(conds, cond => cond(it)));
     },
+    optionsGrouped() {
```

Comment:
Calculated value is actually an array of option groups so let's call it `optionGroups` 💡